src/atomic.rs

```diff
@@ -165,17 +168,18 @@ mod tests {
         }
 
         for region in mem.iter() {
-            iterated_regions.push((region.start_addr(), region.len() as usize));
+            iterated_regions.push(GuestMmapRange::new(
+                region.start_addr(),
+                region.len() as usize,
+                None,
+            ));
         }
         assert_eq!(regions, iterated_regions);
         assert_eq!(mem.num_regions(), 2);
         assert!(mem.find_region(GuestAddress(0x1000)).is_some());
         assert!(mem.find_region(GuestAddress(0x10000)).is_none());
 
-        assert!(regions
-            .iter()
-            .map(|x| (x.0, x.1))
-            .eq(iterated_regions.iter().copied()));
+        assert!(regions.iter().eq(iterated_regions.iter()));
```

Comment:
Isn't that guaranteed by the `assert_eq!(regions, iterated_regions);` above? (I'm not sure about the types here)

Reply:
Yes, you are right.